python_tests/pytest_input_generators.py

```diff
@@ -289,6 +345,169 @@ def define_tensor_error_generator(
         yield SampleInput(input_tensor, **es.kwargs), es.ex_type, es.ex_str
 
 
+def take_along_axis_generator(
+    op: OpInfo, dtype: torch.dtype, requires_grad: bool = False, **kwargs
+):
+    make_arg = partial(
+        make_tensor, device="cuda", dtype=dtype, requires_grad=requires_grad
+    )
+    make_index = partial(
+        make_tensor, device="cuda", dtype=torch.long, requires_grad=False
+    )
+
+    # a.shape, dim, b.shape
+    cases = (
```

Comment:
I'm actually wondering if we have good support on those tho... :confused: 
maybe we don't need to worry about that for now.

Reply:
Since this is opinfo test, we might as well use this opportunity to make it right. Let's add negative dim to the test case. If it breaks, we can either patch the frontend or open an issue to track it.